src/diagnostics.rs

```diff
@@ -52,6 +53,42 @@ pub(crate) fn no_influxdb_identifiers(
     visitor.diagnostics
 }
 
+/// Walk the ast and identifiers that are defined in the script and check that they are
+/// using camelCase rather than snake_case.
+#[derive(Default)]
+struct CamelCaseIdentifierVisitor {
+    diagnostics: Vec<(Option<String>, lsp::Diagnostic)>,
+}
+
+impl<'a> flux::semantic::walk::Visitor<'a>
+    for CamelCaseIdentifierVisitor
+{
+    fn visit(&mut self, node: WalkNode<'a>) -> bool {
+        if let WalkNode::VariableAssgn(assign) = node {
```

Comment:
Why do you assign `node` to variable `assign`? Can I understand it as an ownership transfer?

Reply:
In this case, `node` could be any variant of the `WalkNode` enum. I'm saying "If `node` is a `WalkNode::VariableAssgn`, give me the inner value." In this case, looking at [the declaration of `WalkNode::VariableAssign`](https://github.com/influxdata/flux/blob/master/libflux/flux-core/src/semantic/walk/mod.rs#L63), the inner value is a borrowed [`VariableAssign`](https://github.com/influxdata/flux/blob/master/libflux/flux-core/src/semantic/nodes.rs#L774) instance. No transfer of ownership, only safely unwrapping the enum into its inner value so we can take a look at it.

I briefly considered also checking for [`WalkNode::MemberAssgn`](https://github.com/influxdata/flux/blob/master/libflux/flux-core/src/semantic/walk/mod.rs#L64) but I think that might be more annoying than helpful in that case, as may affect table structures. I think I'll wait to see how this is received before going much further.